app/controllers/extension_controller.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+class ExtensionController < ApplicationController
+  def autorelease_extension
+    @submission = find_submission
+    status_giver = SubmissionStatusGiver.new(@submission)
+    status_giver.can_request_extension?
```

Comment:
Looks like this will prevent the token from working again.  Could we also delete the token somewhere in here?  Just to confirm it can't extend again.